### DISCUSSION TITLE
Best way to have camera image on screen?

### DISCUSSION THREAD

**Discussion by zagi988** [HASwitchPlate/openHASP#589] [Q&A]
Hello everyone.
I'm wondering, what is the best way to have camera image on screen, on demand?
I made a button that runs a Node-Red flow which gets camera image, saves it as .png in /config/www/ folder and then I have an image object that has src:"http://ip:8123/local/image.png" which is generated via openhasp.command by sending jsonl.
But that sometimes gets mi "NO DATA" instead of an image.

So I'm wondering which would be the best way to display an image from a camera?

**Comment by fvanroie** [comment]
What board are you using? Does it have PSram?

**Reply by zagi988** [reply]
Yes, has 8MB.
It's Sunton ESP32-8048S0.

**Reply by fvanroie** [reply]
Please provide the serial log when you set the image `src` attribute.

**Reply by zagi988** [reply]
I'm not using a serial connection nor I have and adapter. I can use telnet, does that help?

**Reply by zagi988** [reply]
`#[16:26:44.596][36852/51152 27][33696/36524  8] MSGR: jsonl={"page":1,"id":200,"obj":"obj","x":0,"y":0,"w":800,"h":480,"radius":0,"bg_grad_dir":0,"bg_opa":255,"bg_color":"black","border_side":0}{"page":1,"id":201,"parentid":200,"obj":"img","x":80,"src":"http://192.168.1.4:8123/local/parking.png","auto_size":1}
#[16:26:46.441][34804/50496 31][35648/36272  2] ATTR: HTTP result -11
#[16:26:46.480][36852/53008 30][34844/35468  2] LVGL: lv_img_design: image source type is unknown
#[16:26:46.491][36852/51096 27][34844/35468  2] LVGL: Image draw: src is NULL`

**Reply by fvanroie** [reply]
HTTP result -11 means HTTPC_ERROR_CONNECTION_REFUSED, so there is an issue getting the image from the server

**Reply by zagi988** [reply]
Yeah but the image is there and when I open it on the PC or mobile phone, no matter how many times I refresh it - always loads. :/

**Reply by smcgann99** [reply]
I don't use HA but I do use NR. I am doing something similar, but I host the file in NR.
If you haven't already setup **httpStatic** then create a directory to store your images and add the path to the NR settings.js file
eg httpStatic: '/home/myusername/node-red-static/. Restart node red and you can access the images there.

**Reply by zagi988** [reply]
Ok, and how do you get it to the panel? push_image or just load it from src:url?

**Reply by hb020** [reply]
-11 can also mean timeout. There is a 1 second timeout now. If the image request takes longer, you will get -11.

In hasp_attribute.cpp, in special_attribute_src(), see  
`http.setTimeout(1000);`
Change that to 
`http.setTimeout(5000);`
, and it might work better.

You may also have a bad image format. To test that, load the png image into local storage, and to show it via "src":"L:/image.png". If that shows, the image format is OK.
It might also be that the web server does not communicate the content-length header. Test that via "inspect" of your favorite browser.

**Comment by smcgann99** [comment]
My setup is **httpStatic: '/home/pi/.node-red/public'** under public I have a number of directories one of which is /home/pi/.node-red/public/images.
Assuming yours was setup the same, then to access this just use `"src":"http://192.168.1.4:1880/images/parking.png"`

**Comment by flexiti** [comment]
call every few sec:
**hasp_process_obj_attribute(MyImageX, "src", "http://192.168.0.90/camera/img/gate.png", true);**  
where gate.png is refreshed by eg. zoneminder or cron
